### PR TITLE
unit tests for LiquidTagGrammar and LineReader

### PR DESCRIPTION
**Pull request by v-nopat**
image

### CODE REVIEW

File: src/server/test/unit/LineReader.test.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+   });
+
+
+   it('LineReader- test getEditedLineContent for 4 line content (edited 4th line)', () => {
+    var TextDocument = { getText: function (range?: Range) {return '<div class="row sectionBlockLayout" style="display: flex; flex-wrap: wrap; padding: 8px; margin: 0px; text-align: center; min-height: auto;">testLin1</div>\n<div>testLin2</div>\n<div>testLin3</div>\n<div>testLin4</div>'},  range:1, _content:'<div class="row sectionBlockLayout" style="display: flex; flex-wrap: wrap; padding: 8px; margin: 0px; text-align: center; min-height: auto;"></div>\n en',
+    offsetAt:1, } as unknown as TextDocument;
+   const manifest= getEditedLineContent(3, TextDocument);
+    expect(manifest).to.be.not.undefined;
+    expect(manifest).eq("<div>testLin4</div>");
+
```
Comment (davidjenni) [inline src/server/test/unit/LineReader.test.ts:89]: style/readability nit: please clean up the extraneous empty lines, e.g. after the last expect(); or the more than one empty line between it() blocks

File: src/server/test/unit/liquidTagGrammer.test.ts [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+import { expect } from 'chai';
+import * as nearley from 'nearley';
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const grammar = require('../../Parser/liquidTagGrammar');
+
+describe('LiquidGrammar', () => {
+
+
+  //#region "Scenarios for Entity Form"
```
Comment (davidjenni) [inline src/server/test/unit/liquidTagGrammer.test.ts:14]: there are no #regions in VSCode. If you want to create structure of cohesive it() clauses, that's what the describe(string) is for

File: src/server/test/unit/liquidTagGrammer.test.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+    expect(liquidExpKeys.next().value).eq('16');
+    expect(parserOutput.get('16')).eq('id');
+ });
+
+  it('webform- grammar should be unambiguous and output length should be 1 only', () => {
+    const parserOutputLength = checkParsedOutputLength("{% webform name: 'test' %}");
+    expect(parserOutputLength).eq(1);
+ });
+
+
```
Comment (davidjenni) [inline src/server/test/unit/liquidTagGrammer.test.ts:225]: nit: yeah, I know I'm nitpicking, please bear with me... as a human code reader, I do favor clean and consistent formatting :-)

**Review by davidjenni — CHANGES_REQUESTED** [review]

File: src/server/test/unit/liquidTagGrammer.test.ts (hunk 3 above)
Comment (v-nopat) [reply src/server/test/unit/liquidTagGrammer.test.ts]: done

File: src/server/test/unit/LineReader.test.ts (hunk 1 above)
Comment (v-nopat) [reply src/server/test/unit/LineReader.test.ts]: done

File: src/server/test/unit/liquidTagGrammer.test.ts (hunk 2 above)
Comment (v-nopat) [reply src/server/test/unit/liquidTagGrammer.test.ts]: _one_

File: src/server/test/unit/LineReader.test.ts (hunk 1 above)
Comment (v-nopat) [reply src/server/test/unit/LineReader.test.ts]: @davidjenni *please review/approve now

File: src/server/test/unit/LineReader.test.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable no-var */
+/*!
```
Comment (davidjenni) [inline src/server/test/unit/LineReader.test.ts:3]: stick with the 2 header formats we already have, don't need a 3rd variation; see e.g. src\server\telemetry\ServerTelemetry.ts or src\client\test\unit\CliAcquisition.test.ts

Also, move the eslint exception comments below the header

File: src/server/test/unit/liquidTagGrammer.test.ts [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+ 
+   it('test valid index value(entityform with attribute name) in Liquid Expression', () => {
+       const parserOutput = parseLiquidAndGetKeyValue("{% entityform name: 'test' %}");
+       expect(parserOutput.size).eq(1);
+       const liquidExpKeys = parserOutput.keys();
+       expect(liquidExpKeys.next().value).eq('21');
+       expect(parserOutput.get('21')).eq('name');
+    });
+
+    it('test valid index value(entityform with attribute ID) in Liquid Expression', () => {
```
Comment (davidjenni) [inline src/server/test/unit/liquidTagGrammer.test.ts:37]: indenting is off, yes, I know, I'm an annoying nit picker. Use VSCode's auto formatting via Ctr+Shift+P, then "Format Selection" or "Format Document"
image

 (you are using VSCode to edit, right :-) )

**Review by davidjenni — CHANGES_REQUESTED** [review]

File: src/server/test/unit/liquidTagGrammer.test.ts (hunk 5 above)
Comment (v-nopat) [reply src/server/test/unit/liquidTagGrammer.test.ts]: done. from format document.

File: src/server/test/unit/LineReader.test.ts (hunk 4 above)
Comment (v-nopat) [reply src/server/test/unit/LineReader.test.ts]: done

**Review by davidjenni — APPROVED** [review]

**Pull request merged**